cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -96,10 +565,65 @@ func (p *AzdoCiProvider) configureConnection(
 	credentials json.RawMessage,
 	console input.Console) error {
 
-	return errors.New("not implemented")
+	azureCredentials, err := parseCredentials(ctx, credentials)
+	if err != nil {
+		return err
+	}
+
+	p.credentials = azureCredentials
+	details := repoDetails.details.(*AzdoRepositoryDetails)
+	org, err := azdo.EnsureOrgNameExists(ctx, p.Env, console)
+	if err != nil {
+		return err
+	}
+	pat, err := azdo.EnsurePatExists(ctx, p.Env, console)
+	if err != nil {
+		return err
+	}
+	connection, err := azdo.GetConnection(ctx, org, pat)
+	if err != nil {
+		return err
+	}
+	err = azdo.CreateServiceConnection(ctx, connection, details.projectId, *p.Env, *p.credentials, console)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// struct used to deserialize service principal json object
+
+// parses the incoming json object and deserializes it to a struct
+func parseCredentials(ctx context.Context, credentials json.RawMessage) (*azdo.AzureServicePrincipalCredentials, error) {
+	azureCredentials := azdo.AzureServicePrincipalCredentials{}
+	if e := json.Unmarshal(credentials, &azureCredentials); e != nil {
+		return nil, fmt.Errorf("setting terraform env var credentials: %w", e)
+	}
+	return &azureCredentials, nil
 }
 
 // configurePipeline create Azdo pipeline
-func (p *AzdoCiProvider) configurePipeline(ctx context.Context) error {
-	return errors.New("not implemented")
+func (p *AzdoCiProvider) configurePipeline(ctx context.Context, repoDetails *gitRepositoryDetails, provisioningProvider provisioning.Options) error {
```

Comment:
Maybe include in the comment that this ends up setting `repoDetails.details.buildDefinition` to the build definition that was created?

Reply:
https://github.com/Azure/azure-dev/issues/756